doc/connectivity/networking/conn_mgr/main.rst

```diff
@@ -0,0 +1,442 @@
+   Only the timeout setting affects this.
+
+   For instance, if a connection attempt on an iface times out, the iface will not attempt to reconnect, even if it is persistent.
+
+   Conversely, if there is not a specified timeout, the iface will try to connect forever until it succeeds, even if it is not persistent.
+
+   See :ref:`conn_mgr_impl_tp_persistence_during_connect` for the equivalent implementation guideline.
+
+.. _conn_mgr_control_events:
+
+Control events
+==============
+
+Connectivity control fires :ref:`network management <net_mgmt_interface>` events to inform the application of important state changes.
+
+See :ref:`conn_mgr_impl_guidelines_fire_events` for the corresponding connectivity implementation guideline.
+
+.. _conn_mgr_control_events_fatal_error:
+
+Fatal Error
+-----------
+
+The :c:macro:`NET_EVENT_CONN_IF_FATAL_ERROR` event is raised whenever an iface encounters an error from which it cannot recover (meaning any subsequent attempts to associate are guaranteed to fail, and all such attempts should be abandoned).
+
+Handlers of this event will be passed a pointer to the iface for which the fatal error occurred.
+Individual connectivity implementations may also pass an application-specific data pointer.
+
+.. _conn_mgr_control_events_timeout:
+
+Timeout
+-------
+
+The :c:macro:`NET_EVENT_CONN_CMD_IF_TIMEOUT` event is raised whenever an :ref:`iface association <conn_mgr_control_operation_connecting>` attempt :ref:`times out <conn_mgr_control_timeouts>`.
+
+Handlers of this event will be passed a pointer to the iface that timed out attempting to associate.
+
+.. _conn_mgr_control_events_listening:
+
+Listening for control events
+----------------------------
+
+You may listen for control events as follows:
+
+.. code-block:: c
+
+   /* Declare a net_mgmt callback struct to store the callback */
+   struct net_mgmt_event_callback my_conn_evt_callback;
+
+   /* Declare a handler to recieve control events */
+   static void my_conn_evt_handler(struct net_mgmt_event_callback *cb,
+                                   uint32_t event, struct net_if *iface)
+   {
+           if (event == NET_EVENT_CONN_IF_TIMEOUT) {
+                   /* Timeout occurred, handle it */
+           } else if (event == NET_EVENT_CONN_IF_FATAL_ERROR) {
+                   /* Fatal error occurred, handle it */
+           }
+
+           /* Otherwise, it's some other event type we didn't register for. */
+   }
+
+   int main()
+   {
+           /* Configure the callback struct to respond to (at least) the CONN_IF_TIMEOUT
+            * and CONN_IF_FATAL_ERROR events.
+            *
+            * Note that the callback may also be fired for events other than those specified here!
+            * (See the net_mgmt documentation)
+            */
+
+           net_mgmt_init_event_callback(
+                   &conn_mgr_conn_callback, conn_mgr_conn_handler,
+				           NET_EVENT_CONN_IF_TIMEOUT | NET_EVENT_CONN_IF_FATAL_ERROR
```

Comment:
Indentation looks funny here, I suggest indenting with spaces in this example

Reply:
Argh, thanks, did my best to get rid of tabs but looks like I missed a line